test/Constraints/closures.swift

```diff
@@ -1042,3 +1042,31 @@ let explicitUnboundResult2: (Array<Bool>) -> Array<Int> = {
 let explicitUnboundResult3: (Array<Bool>) -> Array<Int> = {
   (arr: Array) -> Array in [true]
 }
+
+// SR-13239
+func callit<T>(_ f: () -> T) -> T {
+  f()
+}
+
+func callitArg<T>(_ a: T, _ f: () -> T) -> T { 
+  f()
+}
+
+func testSR13239() -> Int {
+  callit {
```

Comment:
I was looking at these examples and it looks like we try `Int` and `Void` for different variables - `T` and result of a closure, I think we should try to detect whether they actually represent the same thing and score some of the fixes higher e.g. contextual mismatch between result of `testSR13239` and result of a `callit` because its result type is inferred from result of a closure.

Reply:
Ah I think I got the idea, thank you!
I started to look already, maybe I can have something soon :)